app/scripts/analysis/charts/charts.coffee

```diff
@@ -1,86 +1,671 @@
+          #        .attr("transform",(d) -> return "translate(" + x(d.x) + "," + y(d.y) + ")")
+
+
+          bar.append('rect')
+          .style('fill', 'steelblue')
+          .attr('x', (d,i) -> i*5 + x d.x)
+          .attr('y', (d) -> height - y d.y)
+          .attr('width', (d) -> x d.dx)
+          .attr('height', (d) -> y d.y)
+          .on('mouseover', () -> d3.select(this).transition().style('fill', 'orange'))
+          .on('mouseout', () -> d3.select(this).transition().style('fill', 'steelblue'))
+
+          bar.append('text')
+          .attr('x', (d,i) -> i*5 + x d.x)
+          .attr('y', (d) -> height - y d.y)
+          .attr('dx', (d) -> .5*x d.dx)
+          .attr('dy', '20px')
+          .attr('fill', '#fff')
+          .attr('text-anchor', 'middle')
+          .attr('z-index', 1)
+          .text (d) -> d.y
+
+        plotHist(5) #pre-set value of slider
+
+        d3.select('#slider')
+        .on('change', () ->
+          bins = parseInt this.value
+          plotHist(bins)
+        )
+
+      _drawScatterplot = () ->
+
+#        makePairs(data)
+
+
+        x = d3.scale.linear().domain([xMin,xMax]).range([ 0, width ])
+        y = d3.scale.linear().domain([yMin,yMax]).range([ height, 0 ])
+        xAxis = d3.svg.axis().scale(x).orient('bottom')
+        yAxis = d3.svg.axis().scale(y).orient('left')
+
+
+        # values
+        xValue = (d)->parseFloat d.x
+        yValue = (d)->parseFloat d.y
+
+        # map dot coordination
+        xMap = (d)-> x xValue(d)
+        yMap = (d)-> y yValue(d)
+
+        # set up fill color
+        cValue = (d)-> d.y
+        color = d3.scale.category10()
+
+        # x axis
+        _graph.append("g")
+        .attr("class", "x axis")
+        .attr("transform", "translate(0," + height + ")")
+        .call xAxis
+        .append('text')
+        .attr('class', 'label')
+        .attr('x', width)
+        .attr('y', -6)
+        .style('text-anchor', 'end')
+        .text gdata.xLab.value
+
+        # y axis
+        _graph.append("g")
+        .attr("class", "y axis")
+        .call(yAxis)
+        .append("text")
+        .attr('class', 'label')
+        .attr("transform", "rotate(-90)")
+        .attr("y", 6)
+        .attr("dy", ".71em")
+        .style("text-anchor", "end")
+        .text gdata.yLab.value
+
+        # add the tooltip area to the webpage
+        tooltip = container
+        .append('div')
+        .attr('class', 'tooltip')
+        .style('opacity', 0)
+
+        # draw dots
+        _graph.selectAll('.dot')
+        .data(data)
+        .enter().append('circle')
+        .attr('class', 'dot')
+        .attr('r', 5)
+        .attr('cx', xMap)
+        .attr('cy', yMap)
+        .style('fill', (d)->color cValue(d))
+        .on('mouseover', (d)->
+          tooltip.transition().duration(200).style('opacity', .9)
+          tooltip.html('<div style="background-color:white; padding:5px; border-radius: 5px">(' + xValue(d)+ ',' + yValue(d) + ')</div>')
```

Comment:
(style): styles for elements should be defined in .less style file, e.g. see [styles for footer](https://github.com/SOCR/SOCRAT/blob/dev/app/styles/_general.less)